runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,158 @@
+package resolvers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"maps"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/testruntime"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Sources    map[string]yaml.Node
+	Models     map[string]yaml.Node
+	Dashboards map[string]yaml.Node
+	APIs       map[string]yaml.Node
+}
+type Resolvers struct {
+	Project    Project
+	Connectors map[string]*testruntime.InstanceOptionsForResolvers
+	Tests      map[string]*Test
+}
+
+type Test struct {
+	Resolver string
+	Options  runtime.ResolveOptions
+	Result   []map[string]any
+}
+
+var update = flag.Bool("update", false, "Update test results")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	os.Exit(m.Run())
+}
+
+func TestResolvers(t *testing.T) {
+
+	entries, err := os.ReadDir("./")
+	require.NoError(t, err)
+	var reg = regexp.MustCompile(`^(.*)_resolvers_test.yaml$`)
+	for _, e := range entries {
+		if reg.MatchString(e.Name()) {
+			t.Log("Running with", e.Name())
```

Comment:
I believe `t.Run` can be nested any number of times, so maybe use `t.Run` with the filename, and then do a nested call to `t.Run` for the test names. 

Using `t.Run` automatically prints test run status and success/failure info

Reply:
It won't run in vscode https://github.com/golang/vscode-go/issues/3022